mongoexport/main/mongoexport.go

```diff
@@ -91,36 +73,34 @@ func main() {
 		inputOpts.ReadPreference = "nearest"
 	}
 
-	var mode mgo.Mode
-	if opts.ReplicaSetName != "" || isMongos {
-		mode = mgo.Primary
-	} else {
-		mode = mgo.Nearest
```

Comment:
Talked to @xdg about this. Setting the default read preference to nearest was actually trying to make a direct connection to the server. This is already taken care of in the new common library (https://github.com/mongodb/mongo-tools-common/blob/master/options/options.go#L472) so this isn't needed.